### PR TITLE
Added backend CRUD endpoints for Personal Schedules

### LINKED ISSUE #15: 40-backend-crud-PersonalSchedules
Backend CRUD personal schedules

I recommend breaking this issue up into multiple issues.

- [x] In this issue you will create backend Swagger endpoints for crud operations for Personal Schedules.

In this epic, we will not yet reach the point where you will be able to
actually add or delete courses from these schedules; for now they, are just
objects with:

- [x]  Name
- [x] Description
- [x]  Quarter
- [x] And, like the Todos in the sample code for demo-spring-react-example-v2, they *belong to a user*.

So we need the full set of operations where:

- [x]  For admins, admins can manipulate EVERYONE's personal schedules objects
- [x]  For users, they can only manipulate their own.

This is basically just reimplementing the Todo's entities, repositories, and controllers
but with Name/Description/Quarter instead of the fields for Todos.

Add your own ACs, and break this up into multiple issues as needed.

### PR DESCRIPTION
**Pull request by AHuangHe**
# Overview

Implemented backend code for Personal Schedules:
- Personal Schedules ENTITY
- Personal Schedules REPOSITORY
- Personal Schedules CONTROLLER and CONTROLLER TESTS
  - USER -> get all, get, delete, and put/edit (own); any USER can post personal schedules
  - ADMIN -> get all, get, delete, and put/edit (any user)

# Issues Addressed

Closes #15 

# Details
Personal Schedules controller in Swagger UI (in localhost):
cs156_team04

### CODE REVIEW

**Comment by bobhe9606** [comment]
Looks Good to Me!

**Review by bobhe9606 — DISMISSED** [review]
LGTM

File: src/test/java/edu/ucsb/cs156/example/controllers/PersonalSchedulesControllerTests.java [hunk 1]
```diff
@@ -0,0 +1,588 @@
...
+
+    @WithMockUser(roles = { "USER" })
+    @Test
+    public void personal_schedules_all() throws Exception {
+
+        // arrange
+
+        User thisUser = currentUserService.getCurrentUser().getUser();
+
+        PersonalSchedule s1 = PersonalSchedule.builder().name("Schedule 1").description("Schedule 1").quarter("Quarter 1").user(thisUser).id(1L).build();
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/example/controllers/PersonalSchedulesControllerTests.java]: Quarter should always be in the format "YYYYQ" e.g. 
* "20221" for Winter 2022
* "20222" for Spring 2022
* "20223" for Summer 2022
* "20224" for Fall 2022
* "20231" for Winter 2023, etc.

Please change this in the examples quarters (e.g. not `Quarter 1` but something like `20221`)

File: src/test/java/edu/ucsb/cs156/example/controllers/PersonalSchedulesControllerTests.java [hunk 2]
```diff
@@ -0,0 +1,588 @@
...
+        PersonalSchedule s2 = PersonalSchedule.builder().name("Schedule 2").description("Schedule 2").quarter("Quarter 2").user(thisUser).id(2L).build();
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/example/controllers/PersonalSchedulesControllerTests.java]: Same regarding quarter

File: src/test/java/edu/ucsb/cs156/example/controllers/PersonalSchedulesControllerTests.java [hunk 3]
```diff
@@ -0,0 +1,588 @@
...
+    // Tests for /api/PersonalSchedules
+
+    @WithMockUser(roles = { "USER" })
+    @Test
+    public void personal_schedule_exists__user() throws Exception {
+
+        // arrange
+
+        User u = currentUserService.getCurrentUser().getUser();
+        PersonalSchedule s1 = PersonalSchedule.builder().name("Schedule 1").description("Schedule 1").quarter("Quarter 1").user(u).id(7L).build();
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/example/controllers/PersonalSchedulesControllerTests.java]: Same regarding quarter.  Please do a search and replace and make sure that all example values for quarter follow this convention

**Review by pconrad — CHANGES_REQUESTED** [review]
Please see notes about quarter.. .should always be in YYYYQ format.

Also: the title `Ah backend ps` doesn't describe what the PR does.

Please use something more descriptive such as "Backend CRUD endpoints for Personal Schedule".   

Please also consider adding a screenshot of the Swagger interface for the controller, and then deleting the parts of the PR template that no longer apply.

**Comment by edisonnchen** [comment]
Should swagger enforce a proper YYYYQ format?

Screen Shot 2022-03-08 at 7 24 46 PM

Screen Shot 2022-03-08 at 7 25 12 PM

**Review by edisonnchen — CHANGES_REQUESTED** [review]
Swagger doesn't require YYYYQ when inputing date. If that isn't an issue/requirement then the rest LGTM!

EDIT: Seems like the string filtering is done through frontend

**Review by edisonnchen — DISMISSED** [review]
LGTM

**Review by bobhe9606 — DISMISSED** [review]
LGTM

**Review by pconrad — COMMENTED** [review]
This needs to be removed from the PR Description: 

image

**Comment by AHuangHe** [comment]
> This needs to be removed from the PR Description:
> 
> image

Has been deleted

File: src/main/java/edu/ucsb/cs156/example/entities/PersonalSchedule.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Entity(name = "personal_schedules")
+public class PersonalSchedule {
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private long id;
+
+  // This establishes that many todos can belong to one user
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/example/entities/PersonalSchedule.java]: This comment refers to todo, but it should refer to personal schedules.

**Review by pconrad — CHANGES_REQUESTED** [review]
One minor fix to a comment, then LGTM

File: src/main/java/edu/ucsb/cs156/example/entities/PersonalSchedule.java (hunk 4 above)
Comment (AHuangHe) [reply src/main/java/edu/ucsb/cs156/example/entities/PersonalSchedule.java]: Has been changed!

**Review by edisonnchen — APPROVED** [review]
LGTM

**Review by tai-jui — APPROVED** [review]
LGTM!

**Review by andrewhlu — APPROVED** [review]
LGTM! Will merge after test runs complete

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**